src/gpu/intel/jit/gemm/generator/pieces/post_ops.cxx

```diff
@@ -301,28 +299,57 @@ bool BLASKernelGenerator<hw>::gemmBinaryOpC(BinaryOp op, bool row, bool column,
         if (checkRemY)
             cmp(simt | gt | state.flagAP, remY, 0);
 
-        for (int y = 0; y < unrollY; y++) {
-            if (checkRemY) {
-                simtCF ? goto12(16 | ~state.flagAP, lDone)
-                       :   jmpi(1  | ~state.flagAP, lDone);
+        const int grouped_ops = 16;
+        for (int y0 = 0; y0 < unrollY; y0 += grouped_ops) {
+            auto all_CO_regs = std::vector<ngen::GRFRange>(grouped_ops);
+            Label lDoneLoading;
+            // Perform loads
+            for (int i = 0; i < grouped_ops; i++) {
+                const int y = y0 + i;
+                if (y >= unrollY) break;
+                auto &CO_regs = all_CO_regs[i];
+                CO_regs = state.ra.alloc_range(getRegCount(CO_layout));
+                if (checkRemY) {
+                    simtCF ? goto12(16 | ~state.flagAP, lDoneLoading)
+                        :   jmpi(1  | ~state.flagAP, lDoneLoading);
+                }
+                loadMatrix(CO_regs, CO_layout, CO, CO_strategy, CO_addrs, strategy, state);
+                if (recip) map(hw, Tco, CO_regs, CO_regs, strategy, [&](int simd, GRF r, GRF) {
+                    inv(simd, r, r);
+                });
```

Comment:
I'd suggest moving this later -- maybe just before (new) line 336, so you can absorb more load latency for binary div operations too.

Reply:
Thanks, I missed this use of registers during divs, I've moved it into the operate loop as you suggest. However, I moved it to after the goto/jmp block, since that's where it was in the load loop before - do you see an issue with this?